### ISSUE TITLE
Use Omnibar to search through user defined commands, similar to Emacs search for commands

### ISSUE THREAD

**Issue by vlad-terin** [brookhong/Surfingkeys#2088]
## Prelude
* Could you find answer in below documents?

    1. [FAQ](https://github.com/brookhong/Surfingkeys/wiki/FAQ) / [常见问题](https://github.com/brookhong/Surfingkeys/wiki/%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98)
    1. [README](https://github.com/brookhong/Surfingkeys/blob/master/README.md) / [读我](https://github.com/brookhong/Surfingkeys/blob/master/README_CN.md)

Couldn't find it.
* Have you searched your problem in [issues](https://github.com/brookhong/Surfingkeys/issues)?

Yes.
## Error details

Not an error, simply trying to figure out a way to search through user defined searchengines within an  Omnibar and search through annotations, rather than a keybinding,  similar how Emacs allows to search for commands. 

I tried defining SearchEngine parameters in Front.openOmnibar, but couldn't get it to work. 

Maybe someone has a graceful solution to this already. 

## Context

**browsing anywhere, deciding to go to a search engine i defined. I have many of them and forget some by not usign them often. My current process is opening shortcut helper, search through annotations, find the key to it, and then execute the key. Would like to have the omnibar be openeded to search through user defined aliases.**

**Comment by Shirkit** [comment]
I know it's not the same, but you can press *?* to for other commands not related to search engines.

**Comment by vlad-terin** [comment]
Yes, that's my way of findings search engines I forgot, but it would be nice to be able to bind it to a command that can
a) search by annotations
b) execute on enter

**Comment by Fred-Vatin** [comment]
I think we need to add an omnibar type: **SearchEngines** (plural).

Instead of using omnibar tabs or bookmarks or else then typing an alias fallowed by `<Space>` to switch to search with given engine, I suggest an omnibar that list all defined search engines that we could filter and select.
The list would present any engine like: `[icon] [alias] [engine name]` with different color for alias and name. We could select then `<Enter>` or type the alias then `<Enter>`.

I suggest `addSearchAlias` to add an optional fallback url parameter in options object. It will be used when the user uses the search engine with no input.
